de.fu_berlin.inf.dpp.core/test/junit/de/fu_berlin/inf/dpp/versioning/JGitServiceTest.java

```diff
@@ -0,0 +1,69 @@
+package de.fu_berlin.inf.dpp.versioning;
+
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.Status;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class JGitServiceTest {
+
+    public enum RepositoryOwner {
+        Host, Client
+    }
+
+    @Before
+    public void setUp() {
+    }
+
+    public static Git getSettedUpRepo(RepositoryOwner owner)
+        throws IllegalStateException, GitAPIException, IOException {
+
+        @Rule
+        TemporaryFolder tempFolder = new TemporaryFolder();
+
+        File tempSubfolder = tempFolder.newFolder("tempFolder"
+            + owner.toString());
+        File tempFile = tempSubfolder.createTempFile("tempFile", ".txt");
+        Git git = Git.init().setDirectory(tempFile.getParentFile()).call();
+        git.add().addFilepattern(tempFile.getPath()).call();
+        git.commit().setMessage("Initial commit").call();
+        git.tag().setName("ClientCheckout").setAnnotated(false)
+            .setForceUpdate(true).call();
+
+        // the Host will be 1 commit forward
+        if (owner == RepositoryOwner.Host) {
```

Comment:
Instead of creating a method that contains the logic for creating one setup for Client and one for the Host I would move the logic into the test and write a generic `getSettedUpRepo` version:
Depending on your requirements you could do something like:
* getSettedUpRepo(Integer commitNr) - method that creates a repo with as much commits as defined in commitNr
* getSettedUpRepo(List<String> commits) - method that creates a repo with as much commits as defined in by the list commits and uses the values as commit messages
* getSettedUpRepo(LinkedHashMap<String, List<File>> commits) -  method that creates a repo with as much commits as defined in by the map. The map contains the commit message and the files to be commited
...